Cython/Compiler/ModuleNode.py

```diff
@@ -2759,8 +2780,18 @@ def generate_main_method(self, env, code):
                 main_method=Options.embed,
                 wmain_method=wmain))
 
+    def punycode_module_name(self, name):
+        # from PEP483
+        try:
+            name = b'_' + name.encode('ascii')
```

Comment:
I don't think this underscore belongs into a method that claims to return the `punycode_module_name`.
Maybe the method should accept an additional `prefix` and then `build…` something?